tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/11_carrierRestriction.js

```diff
@@ -0,0 +1,335 @@
+
+// import test context
+const testContext = require('@utils/testContext');
+
+const baseContext = 'functional_BO_catalog_discounts_cartRules_CRUDCartRule_carrierRestriction';
+
+// Import expect from chai
+const {expect} = require('chai');
+
+let browserContext;
+let page;
+
+const cartRuleCode = new CartRuleFaker(
+  {
+    name: 'addCartRuleName',
+    code: '4QABV6L3',
+    carrierRestriction: true,
+    discountType: 'Amount',
+    discountAmount: {
+      value: 10,
+      currency: 'EUR',
+      tax: 'Tax included',
+    },
+  },
+);
+
+describe('BO - Catalog - Cart rules : Case 11 - Carrier Restriction', async () => {
+  // before and after functions
+  before(async function () {
+    browserContext = await helper.createBrowserContext(this.browser);
+    page = await helper.newTab(browserContext);
+  });
+
+  after(async () => {
+    await helper.closeBrowserContext(browserContext);
+  });
+
+  describe('Create a Cart Rules', async () => {
+    it('should login in BO', async function () {
+      await loginCommon.loginBO(this, page);
+    });
+
+    it('should go to \'Catalog > Discounts\' page', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'goToDiscountsPage',
+        baseContext,
+      );
+
+      await dashboardPage.goToSubMenu(
+        page,
+        dashboardPage.catalogParentLink,
+        dashboardPage.discountsLink,
+      );
+
+      const pageTitle = await cartRulesPage.getPageTitle(page);
+      await expect(pageTitle).to.contains(cartRulesPage.pageTitle);
+    });
+
+    it('should go to new cart rule page', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'goToNewCartRulePage',
+        baseContext,
+      );
+
+      await cartRulesPage.goToAddNewCartRulesPage(page);
+
+      const pageTitle = await addCartRulePage.getPageTitle(page);
+      await expect(pageTitle).to.contains(addCartRulePage.pageTitle);
+    });
+
+    it('should create cart rule', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'createCartRule',
+        baseContext,
+      );
+
+      const validationMessage = await addCartRulePage.createEditCartRules(page, cartRuleCode);
+      await expect(validationMessage).to.contains(addCartRulePage.successfulCreationMessage);
+    });
+
+    it('should view my shop', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        'viewMyShop1',
+        baseContext,
+      );
+
+      // View my shop and init pages
+      page = await addCartRulePage.viewMyShop(page);
+
+      await foHomePage.changeLanguage(page, 'en');
+      const isHomePage = await foHomePage.isHomePage(page);
+      await expect(isHomePage, 'Fail to open FO home page').to.be.true;
+    });
+  });
+
+  const tests = [
+    {args: {testIdentifier: 'cartRuleAccepted', testTitle: 'for the first time'}},
```

Comment:
Do you have another missed argument ?
Maybe you should remove the foreach

Reply:
Done